crowbar_framework/app/models/ntp_service.rb

```diff
@@ -62,18 +62,20 @@ def transition(inst, name, state)
     #
     if state == "discovered"
       @logger.debug("NTP transition: discovered state for #{name} for #{state}")
-      db = Proposal.where(barclamp: "ntp", name: inst).first
       role = RoleObject.find_role_by_name "ntp-config-#{inst}"
+      node = NodeObject.find_node_by_name name
+      result = false
 
-      if role.override_attributes["ntp"]["elements"]["ntp-server"].nil? or
-         role.override_attributes["ntp"]["elements"]["ntp-server"].empty?
-        @logger.debug("NTP transition: make sure that ntp-server role is on first: #{name} for #{state}")
-        result = add_role_to_instance_and_node("ntp", inst, name, db, role, "ntp-server")
-      else
-        node = NodeObject.find_node_by_name name
-        unless node.role? "ntp-server"
-          @logger.debug("NTP transition: make sure that ntp-client role is on all nodes but first: #{name} for #{state}")
-          result = add_role_to_instance_and_node("ntp", inst, name, db, role, "ntp-client")
+      Proposal.where(barclamp: "ntp", name: inst).first do |db|
+        if role.override_attributes["ntp"]["elements"]["ntp-server"].nil? ||
+            role.override_attributes["ntp"]["elements"]["ntp-server"].empty?
+          @logger.debug("NTP transition: make sure that ntp-server role is on first: #{name} for #{state}")
```

Comment:
Metrics/LineLength: Line is too long. [107/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)